specs/_features/maxeb_increase/capella.py

```diff
@@ -1776,6 +1788,44 @@ def process_execution_layer_withdraw_request(
     ))
 
 
+def is_consolidated(validator: Validator) -> bool:
+   validator.consolidated_to != UNSET_CONSOLIDATED_TO
+
+
+def resolve_slashed_index(state: BeaconState, index: ValidatorIndex) -> ValidatorIndex:
+    if is_consolidated(state.validators[index]):
+        # Recursively resolve consolidated index
+        return resolve_slashed_index(state, state.validators[index].consolidated_to)
+    else:
+        return index
+
+
+def process_consolidation(state: BeaconState, consolidation: Consolidation) -> None:
+    target_validator = state[consolidation.target_index]
+    source_validator = state[consolidation.source_index]
+
+    assert is_active_validator(target_validator) and not target_validator.slashed and not is_consolidated(target_validator)
+    assert not source_validator.slashed and not is_consolidated(source_validator)
+
+    # verify source withdrawal credentials, which have authority over validating keys
+    assert source_validator.withdrawal_credentials[:1] == ETH1_ADDRESS_WITHDRAWAL_PREFIX
+    assert source_validator.withdrawal_credentials[12:] == consolidation.source_address
+
+    # target validator accepts consolidation operation onto itself
+    # Fork-agnostic domain since address changes are valid across forks
+    domain = compute_domain(DOMAIN_CONSOLIDATION, genesis_validators_root=state.genesis_validators_root)
+    signing_root = compute_signing_root(source_validator.pubkey, domain)
+    assert bls.Verify(target_validator.pubkey, signing_root, consolidation.target_signature)
+
+    state.balances[consolidation.target_index] += state.balances[consolidation.source_index]
+    state.balances[consolidation.source_index] = 0
+
+    source_validator.consolidated_to = consolidation.target_index
+    # Balance is not exiting the active set, do not apply churn
+    source_validator.exit_epoch = get_current_epoch(state)
```

Comment:
```suggestion
    source_validator.exit_epoch = compute_activation_exit_epoch(get_current_epoch(state))
```

Reply:
Why should this validator with 0 balance remain active?

Reply:
You're right. In the case when we set the balance to 0 right away we should deactivate a validator instantly. I was thinking about the case when we gracefully make consolidated validator exit which is helpful in terms of its duties to be shut down gracefully considering shuffling lookahead